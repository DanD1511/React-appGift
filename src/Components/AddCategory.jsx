import React, { useState } from 'react'
import PropTypes from 'prop-types';


export default function AddCategory({onNewCategory,borrarGifs}) {
    
    const [inputValue, setInputValue] = useState('')

    //Evento del input
    const onInputChange=(evento)=>{
        //InputValue Almacena el valor del inputValue
        setInputValue(evento.target.value)
    }

    //Evento del form
    const onSubmit=(evento)=>{
        evento.preventDefault()
        let newCategory=inputValue
        //Si viene vacio
        if(inputValue.trim().length >1){
            //Recordar que el primer valor de la funcion es [...categories] y luego se le agrega el inputValue
            onNewCategory(newCategory)
            //Borramos el contenido del input
            setInputValue('')
            
        }
        else{
            return
        }
    }
  return (
    <>
    
    <form action="" onSubmit={onSubmit} aria-label='form'>

        <input type="text" placeholder="Insert an category" onInput={onInputChange} value={inputValue} />
    </form>
    </>
  )
}

AddCategory.propTypes={
    onNewCategory:PropTypes.func.isRequired
}