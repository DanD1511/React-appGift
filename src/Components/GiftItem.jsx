import React from 'react'
import PropTypes from 'prop-types'; // ES6

export const GiftItem = ({id,title,urlImg}) => {
  return (
        <>
        <div className='card'>
            <img src={urlImg} alt={title} />
            <p>{title}</p>  
        </div>
        
        </>
  )
}

GiftItem.propTypes={
  title:PropTypes.string.isRequired,
  urlImg:PropTypes.string.isRequired
}