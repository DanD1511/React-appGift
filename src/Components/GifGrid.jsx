
import { GiftItem } from "./GiftItem"
import { useFetchGif } from "../hooks/useFetchGif"
import PropTypes from 'prop-types'; // ES6



export const GifGrid = ({category}) => {
    
    //Usamos el custom hooks que creamos
    const {gifs,cargando}=useFetchGif(category)
    // console.log(cargando)
    
  return (
    
    <>
        <h3>{category}</h3>
        
        <div className="card-grid">

            {/* Si cargando est√° en true, muestra el mensaje de : Cargando... */}
            {cargando && (<h3>Cargando...</h3>)}
            {/* //Pasamos la info  a un nuevo componente que lo muestre */}
            {gifs.map(({id,title,urlImg})=>(
                <GiftItem key={id} id={id} title={title} urlImg={urlImg}></GiftItem>
            ))}

        </div>

        
    </>
  )
}

GifGrid.propTypes={
  category:PropTypes.string.isRequired
}