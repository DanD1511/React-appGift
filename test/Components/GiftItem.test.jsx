import { render, screen } from "@testing-library/react";
import { GiftItem } from "../../src/Components/GiftItem";

describe('Pruebas en <GiftItem></GiftItem>', () => {
    const title='Dragon ball'
    const urlImg='https://dragonball.com/goku.jpg'

    test('Debe hacer match con el snapshot', () => {
        
        const {container}=render(<GiftItem title={title} urlImg={urlImg}></GiftItem>)
        expect(container).toMatchSnapshot()
    });

    test('Debe mostrar el nombre de la img y su ALT', () => {
        render(<GiftItem title={title} urlImg={urlImg}></GiftItem>)
        //Obtenemos el src y el alt de la imagen
        
        // expect(screen.getByRole('img').src).toBe(urlImg)
        // expect(screen.getByRole('img').alt).toBe(title)

        //Otra manera 
        const {src,alt}=screen.getByRole('img')
        expect(src).toBe(urlImg)
        expect(alt).toBe(title)
    });

    test('Debe de haber un texto <p>{title}</p> en el componente', () => {
        render(<GiftItem title={title} urlImg={urlImg}></GiftItem>)
        //Obtenemos el texto que tenga valor title y verificamos que exista
        //Recordar que screen es una copia del componente
        expect(screen.getByText(title)).toBeTruthy()

    });
});