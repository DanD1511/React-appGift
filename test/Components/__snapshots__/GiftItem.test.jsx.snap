// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem></GiftItem> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon ball"
      src="https://dragonball.com/goku.jpg"
    />
    <p>
      Dragon ball
    </p>
  </div>
</div>
`;
